### ISSUE TITLE
infraction epoch seems to be counted as 1 while being at epoch 0 (possible theoretical flaw)

### ISSUE THREAD

**Issue by McDaan** [anoma/namada#2632]
For example, this validator has infractions for double-signing committed at epoch 0, so according to the Specs (https://specs.namada.net/modules/proof-of-stake/cubic-slashing), he is able to unjail at epoch 6 in theory, but he can't.

Specs mention:
`infraction_epoch + unbonding_len + window_width + 1`
According to SE values on this testnet:
`0 + 4 + 1 + 1 = 6`

```
namadac slashes --validator tnam1q84tqxcm0l792cvtx8ufe6erx00fwwjtggv925l4
Processed slashes:
Infraction epoch 0, block height 5, type Duplicate vote, rate 1
Infraction epoch 0, block height 5, type Duplicate vote, rate 1
Infraction epoch 0, block height 5, type Duplicate vote, rate 1
No enqueued slashes found for tnam1q84tqxcm0l792cvtx8ufe6erx00fwwjtggv925l4
```
```
namadac epoch
Last committed epoch: 6
```
```Transaction added to mempool.
Wrapper transaction hash: F4C3445ACE9B6675A4249A9ACC7B288D0E4FCA9ECED53E3A7CFA980EAA8B40F8
Inner transaction hash: E74ABFF75F4C27903EE92BF466C503B73445BA9D908B4AFDDDAD7E6E1ECEE28A
Wrapper transaction accepted at height 24224. Used 19 gas.
Waiting for inner transaction result...
Transaction was rejected by VPs: [
  "tnam1q84tqxcm0l792cvtx8ufe6erx00fwwjtggv925l4"
].
```
Link to the message: [Namada Discord server](https://discord.com/channels/833618405537218590/1192974425423892550/1207776727749754931)

Perhaps what happens is that epoch 0 is being counted as it was 1?

Following that logic: `epoch 0 as 1+unbonding_length+cubic_slasing_window_len+1 = 7 epochs`

**Comment by brentstone** [comment]
I do not believe this is occurring anywhere in the codebase. There are validators who were slashed the same epoch that have been able to successfully unjail, anyway. Feel free to open another issue if there is something in the code that looks suspicious to you, though. You can look in the `proof_of_stake` crate and at some functions like:
- `namada_apps::node::ledger::shell::Shell::record_slashes_from_evidence`
- `proof_of_stake::slashing::slash`
- `proof_of_stake::slashing::process_slashes`
- `proof_of_stake::unjail_validator`

**Comment by brentstone** [comment]
Furthermore, the user in the Discord message you linked seems to have solved his issue by fixing his keys. Going to close this now.
